src/Monolog/Handler/Slack/SlackRecord.php

```diff
@@ -0,0 +1,249 @@
+     * @var bool
+     */
+    private $includeContextAndExtra;
+
+    /**
+     * @var FormatterInterface
+     */
+    private $formatter;
+
+    /**
+     * @var LineFormatter
+     */
+    private $lineFormatter;
+
+    public function __construct($channel = null, $username = 'Monolog', $useAttachment = true, $iconEmoji = null, $useShortAttachment = false, $includeContextAndExtra = false, FormatterInterface $formatter = null)
+    {
+        $this->channel = $channel;
+        $this->username = $username;
+        $this->iconEmoji = trim($iconEmoji, ':');
+        $this->useAttachment = $useAttachment;
+        $this->useShortAttachment = $useShortAttachment;
+        $this->includeContextAndExtra = $includeContextAndExtra;
+        $this->formatter = $formatter;
+
+        if ($this->includeContextAndExtra) {
+            $this->lineFormatter = new LineFormatter();
+        }
+    }
+
+    public function getSlackData(array $record)
+    {
+        $dataArray = array(
+            'username'    => $this->username,
+            'text'        => '',
+            'attachments' => array(),
+        );
+
+        if ($this->channel) {
+            $dataArray['channel'] = $this->channel;
+        }
+
+        if ($this->formatter) {
+            $message = $this->formatter->format($record);
+        } else {
+            $message = $record['message'];
+        }
+
+        if ($this->useAttachment) {
+            $attachment = array(
+                'fallback' => $message,
+                'text'     => $message,
+                'color'    => $this->getAttachmentColor($record['level']),
+                'fields'   => array(),
+            );
+
+            if ($this->useShortAttachment) {
+                $attachment['title'] = $record['level_name'];
+            } else {
+                $attachment['title'] = 'Message';
+                $attachment['fields'][] = $this->generateAttachmentField('Level', $record['level_name'], true);
+            }
+
+            if ($this->includeContextAndExtra) {
+                foreach (array('extra', 'context') as $key) {
+                    if (empty($record[$key])) {
+                        continue;
+                    }
+
+                    if ($this->useShortAttachment) {
+                        $attachment['fields'][] = $this->generateAttachmentField(
+                            ucfirst($key),
+                            $this->stringify($record[$key]),
+                            true
+                        );
+                    } else {
+                        // Add all extra fields as individual fields in attachment
+                        $attachment['fields'] = array_merge(
+                            $attachment['fields'],
+                            $this->generateAttachmentFields($record[$key])
+                        );
+                    }
+                }
+            }
+
+            $dataArray['attachments'] = array($attachment);
```

Comment:
Removed `json_encode` here. I think this will break `SlackHandler`, despite working fine with `SlackWebhookHandler`. `SlackHandler` sends data with `application/x-www-form-urlencoded` headers, and it looks like slack expects `attachments` field to be json_encoded in that case.

A few optoins I figured:
- json_encode `attachments` in `SlackHandler`
- change content-type to `application/json` and json_encode the whole record in `SlackHandler`

I like second more. What do you think @Seldaek @hkdobrev ?